tensorflow_encrypted/config.py

```diff
@@ -154,35 +171,45 @@ def to_dict(self) -> Dict:
 
     @property
     def players(self) -> List[Player]:
-        return self._players.values()
+        return list(self._players.values())
 
-    def get_player(self, name:str) -> Player:
+    def get_player(self, name: str) -> Player:
         return self._players[name]
 
-    def get_players(self, names:Union[List[str], str]) -> List[Player]:
+    def get_players(self, names: Union[List[str], str]) -> List[Player]:
         if isinstance(names, str):
             names = [name.strip() for name in names.split(',')]
         assert isinstance(names, list)
         return [player for name, player in self._players.items() if name in names]
 
-    def server(self, name:str) -> tf.train.Server:
+    def server(self, name: str) -> tf.train.Server:
         player = self.get_player(name)
         cluster = tf.train.ClusterSpec({self._job_name: self._hostmap})
-        return tf.train.Server(
+        server = tf.train.Server(
             cluster,
             job_name=self._job_name,
             task_index=player.index
         )
+        self._target = server.target
+        print("Hi I'm node %s listening on %s with device_name %s" % (name, server.target, player.device_name))
+        return server
+
+    def session(self, log_device_placement: bool = False) -> tf.Session:
+        global __TFE_DEBUG__
 
-    def session(self, log_device_placement:bool=False) -> tf.Session:
         config = tf.ConfigProto(
             log_device_placement=log_device_placement,
             allow_soft_placement=False,
         )
-        return tf.Session(
+        print('Starting session on target: %s' % self._target, config)
```

Comment:
are all these only meant for debugging right now or would we like them to stay?

Reply:
I added this one for debugging at first, but I find it quite useful overall in the end. It feels like when someone starts a web server.